src/test/ui/regions/regions-nested-fns-2.nll.stderr

```diff
@@ -2,15 +2,13 @@ error[E0597]: `y` does not live long enough
   --> $DIR/regions-nested-fns-2.rs:18:25
    |
 LL |         |z| {
-   |         --- value captured here
+   |           - has type `&'0 isize`
```

Comment:
@davidtwco should the label span be padded to `z` here?

Reply:
It should. I'm aware that the span for closures are off.

Reply:
Can you add a test with a closure having multiple arguments with longer names? I want to see if it is an off-by-one error in the closure arg spans, or if it is that all args are just pointing at the last span of the argument list. It's probably a bug in the parser generating the AST with incorrect spans _or_ a bug in the lowering of the closure AST.

Reply:
@estebank I've updated a test here with a multiple argument closure example:

https://github.com/rust-lang/rust/blob/1373043ca1f7af15b102fda382cfb0e787e254c6/src/test/ui/nll/issue-52534.stderr#L11-L19

Reply:
Thanks! I see it. That sounds to me as somewhere in the parser using `self.span` instead of `self.prev_span`. This is not an uncommon occurrence when expanding diagnostics exposing spans that haven't been used before.

I'll go as far as to say that the following line is incorrect and should be using `self.prev_span`:

https://github.com/rust-lang/rust/blob/f1aefb48d2ec7ac38a66c964396a5aec729b7a28/src/libsyntax/parse/parser.rs#L1860

Could you try that out?

Reply:
@estebank That did the trick, thanks! Pushed a commit with that.